Proposal: Add templates for Issues and PRs (#2611)
paritytech/rust-libp2p: .github/pull_request_template.md

# Description

<!-- Please write a summary of your changes and why you made them.-->

## Links to any relevant issues

<!-- Reference any related issues.-->

> ## Type of change
> 
> <!-- Choose a type of change, and delete any options that are not relevant.-->
> 
> - [ ] Bug Fix
> - [ ] Enhancement
> - [ ] Feature
> - [ ] Refactor
> - [ ] Documentation Fix
> - [ ] Other:

Reviewer: ```suggestion
```

Is this needed? We don't categorize our changelog entries today, thus I am not sure this would be needed.

## Open Questions

<!-- Unresolved questions, if any. -->

## Change checklist

> <!-- Please add a Changelog entry in the appropriate packages. See <https://github.com/libp2p/rust-libp2p/blob/master/docs/release.md#development-between-releases>-->

Reviewer: ```suggestion
<!-- Please add a Changelog entry in the appropriate crates and bump the crate versions if needed. See <https://github.com/libp2p/rust-libp2p/blob/master/docs/release.md#development-between-releases>-->
```

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

> - [ ] A changelog entry has been made in the appropriate packages

Reviewer: ```suggestion
- [ ] A changelog entry has been made in the appropriate crates
```

For the sake of consistency?!
